R/assocTestSingle.R

```diff
@@ -4,7 +4,7 @@ setGeneric("assocTestSingle", function(gdsobj, ...) standardGeneric("assocTestSi
 ## do we want to make imputing to the mean optional?
 setMethod("assocTestSingle",
           "SeqVarIterator",
-          function(gdsobj, null.model, test=c("Score", "Wald"), GxE=NULL, sparse=TRUE, imputed=FALSE, male.diploid=TRUE, genome.build=c("hg19", "hg38"), verbose=TRUE) {
+          function(gdsobj, null.model, test=c("Score", "Wald"), SPA = FALSE, GxE=NULL, sparse=TRUE, imputed=FALSE, male.diploid=TRUE, genome.build=c("hg19", "hg38"), verbose=TRUE) {
```

Comment:
@smgogarten what are your thoughts on having `SPA` as a logical argument here? Alternatively, we could just have it as another option for `test`. The reason I did it this way as that using the `SPA` option still actually does a score test, it just does some p-value adjustment afterwards.

Reply:
Are we planning on dropping the "test" argument entirely and only doing score tests? If so, then I think the "SPA" logical makes sense here.

Reply:
In the short term, I think yes. But a "test" argument might be the best way to incorporate the CMP and BinomiRare tests. 

One thing with SPA is that it first runs the exact same score test code, then does post processing (as opposed to binomirare and cmp which have their own testing code). Additionally, the way it's set up now, using SPA will return both the original `Score.pval` and a new `SPA.pval`. I didn't see a reason to throw out the `Score.pval`, but I suppose we could. That would also address my question below.

Reply:
Is it possible to combine SPA and BinomiRare or CMP? If the SPA=T/F argument only makes sense for one of the `test` options, then it might make more sense to have it be its own `test` option. But I could go either way.

Reply:
From looking at the BR and CMP code, it appears they have the option to just return the score pvalue if it is > 0.05 (the `calc.score` argument seems to determine this; if `TRUE`, get score pvalue, keep if > 0.05, compute BR pval if < 0.05; if `FALSE`, just compute the BR pval for every variant). This is actually similar to the way SPA is currently implemented with `pval.thresh`. **It seems to me that BR and CMP are alternatives to SPA.** 

Given this parallelism, perhaps we should treat them all similarly. We could have a `test` argument with values "Score", "Score.SPA", "BinomiRare", and "CMP".  

We could also then have a common `pval.thresh` (or better name) parameter that applies to "Score.SPA", "BinomiRare", and "CMP", giving the threshold for which you re-compute the p-value using the more sophisticated method. Setting the value to 1 would apply the method to all variants. (Or we could just set this to 0.05 and not make it user defined.)

Reply:
I named this argument `recalc.pval.thresh`. Do you like that or can you think of a better name?

Reply:
I think I like this name. I've been having a hard time coming up with one that I like. 

Once we include the BinomiRare and CMP tests, what would the description be? Something like: "When test != 'Score', variants with a Score p-value below this threshold will have their p-value recalculated using the specified test"

Reply:
The description is currently "If test is not "Score", only recalculate p-values below this threshold." which seems general. But we can update it later to specifically mention the mulltiple test possibilities.